### PR TITLE
feature(aens)!: don't limit pointer keys

### LINKED ISSUE #1240: AENS pointers: allow custom keys
Although the protocol allows to set different pointers for each valid type (e.g. it is possible to set keys for multiple accounts) the SDK currently doesn't provide a possibility to do so.

Currently it is only possible to provide an array of values (e.g. `['ak_...', 'ct_...', 'ok_...', 'ch_...']`) where the respective `default key` is determined automatically.

I propose to change that so that developers can (no-breaking change) **OR** have to (breaking change) provide a key-value map with valid types as values.

I think @hanssv said that there is a limit of maximum pointers that can be set. Not sure about the maximum amount right now.

### PR DESCRIPTION
**Pull request by davidyuk**
closes #1240

### CODE REVIEW

File: src/tx/builder/schema.js [hunk 1]
```diff
@@ -42,12 +42,6 @@ export const NAME_FEE_BID_INCREMENT = 0.05 // # the increment is in percentage
 export const NAME_BID_TIMEOUT_BLOCKS = 480 // # ~1 day
 export const NAME_MAX_LENGTH_FEE = 31 // # this is the max length for a domain that requires a base fee to be paid
 export const NAME_BID_MAX_LENGTH = 12 // # this is the max length for a domain to be part of a bid
-export const NAME_ID_KEY = {
```
Comment (marc0olo) [inline src/tx/builder/schema.js:45]: not sure if we need to remove those keys as these are the default keys. maybe we could have a helper function instead to get the default key for the respective type.

**Review by marc0olo — COMMENTED** [review]
got a question/suggestions regarding the default pointer keys.

could you please also add a test where a custom pointer key independent of the default keys is used? and e.g. a set of all default pointers including the custom pointer is being set and validated?

File: src/tx/builder/schema.js (hunk 1 above)
Comment (davidyuk) [reply src/tx/builder/schema.js]: I've reverted classify function that is doing that: https://github.com/aeternity/aepp-sdk-js/blob/db6ca4c3d89eedf7fa326bd6d0484ea0d94de19e/src/tx/builder/helpers.js#L251-L269

Comment (marc0olo) [reply src/tx/builder/schema.js]: mhh, I don't think we are talking about the same here. not sure. let me ask a question like this:

- how does the sdk user know about the default pointer-keys that have a special meaning? now he would need to know them, right? shouldn't there be an easy way to look those keys with special meaning up?

it's only about those 4 keys

Comment (marc0olo) [reply src/tx/builder/schema.js]: take a look at this example to know what I mean:
- https://github.com/aeternity/aepp-sdk-js/issues/1332#issue-1061841739

Comment (davidyuk) [reply src/tx/builder/schema.js]: > how does the sdk user know about the default pointer-keys that have a special meaning?

I think the best here is to have it documented. As a developer, I see that each pointer has a name, and looks reasonable to go to documentation to check the default pointer names.

`classify` can be used this way:
```js
const ct = 'ct_.....';
sdk.aensUpdate(name, { [classify(ct)]: ct, myPointer: ct });
```

Comment (marc0olo) [reply src/tx/builder/schema.js]: @davidyuk I think we should rename `classify` if it is only used for this purpose. I would prefer sth. like `getDefaultPointerKey` but I am also open for other proposals. I think a general helper function should be more expressive.

I would also expect this to be in the usage guide for the default pointers and additionally 1 custom pointer to make the usage more clear with some small description

**Review by marc0olo — APPROVED** [review]
looks good to me know, thanks @davidyuk! :)

**Review by mradkov — APPROVED** [review]

**Pull request merged**